### PR TITLE
New Resources and Data Sources: aws_secretsmanager_secret*

### LINKED ISSUE #4060: Feature Request: Support AWS Secrets Manager Secrets
Amazon has released a new service, AWS Secrets Manager, which is a secrets management service. https://aws.amazon.com/about-aws/whats-new/2018/04/introducing-aws-secrets-manager/

Please vote on this issue by adding a :+1: reaction to the original issue to help prioritize interest. If you're interested in implementing this feature request, please comment below.

Prerequisites:
* aws-sdk-go v1.13.28: #4048
* `meta.(*AWSClient).acmpcaconn`: #4049

### Terraform Version

terraform 0.10+

### New Resource(s)

- `aws_secretsmanager_secret`

### Terraform Configuration Files

```hcl
# Example implementation, may change when developed
resource "aws_secretsmanager_secret" "example" {
  description = "" # optional
  kms_key_id  = "" # optional
  name        = "" # required
  # ... likely attributes for generating a secret ...

  tags = { # optional
  	key = value
  }
}
```

### Expected Behavior

Resource creates, updates, and deletes Secret Manager secrets.

### Actual Behavior

New feature.

### References

* https://docs.aws.amazon.com/secretsmanager/latest/userguide/
* https://docs.aws.amazon.com/sdk-for-go/api/service/secretsmanager/#SecretsManager.CreateSecret

### PR DESCRIPTION
**Pull request by bflad**
Closes #4060

New Resources:
* `aws_secretsmanager_secret`
* `aws_secretsmanager_secretversion`

New Data Sources:
* `aws_secretsmanager_secret`
* `aws_secretsmanager_secretversion`

```
make testacc TEST=./aws TESTARGS='-run=AwsSecretsManager'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=AwsSecretsManager -timeout 120m
=== RUN   TestAccDataSourceAwsSecretsManagerSecret_Basic
--- PASS: TestAccDataSourceAwsSecretsManagerSecret_Basic (5.54s)
=== RUN   TestAccDataSourceAwsSecretsManagerSecret_ARN
--- PASS: TestAccDataSourceAwsSecretsManagerSecret_ARN (11.24s)
=== RUN   TestAccDataSourceAwsSecretsManagerSecret_Name
--- PASS: TestAccDataSourceAwsSecretsManagerSecret_Name (11.82s)
=== RUN   TestAccDataSourceAwsSecretsManagerSecretVersion_Basic
--- PASS: TestAccDataSourceAwsSecretsManagerSecretVersion_Basic (14.41s)
=== RUN   TestAccDataSourceAwsSecretsManagerSecretVersion_VersionID
--- PASS: TestAccDataSourceAwsSecretsManagerSecretVersion_VersionID (14.18s)
=== RUN   TestAccDataSourceAwsSecretsManagerSecretVersion_VersionStage
--- PASS: TestAccDataSourceAwsSecretsManagerSecretVersion_VersionStage (13.31s)
=== RUN   TestAccAwsSecretsManagerSecret_Basic
--- PASS: TestAccAwsSecretsManagerSecret_Basic (12.01s)
=== RUN   TestAccAwsSecretsManagerSecret_Description
--- PASS: TestAccAwsSecretsManagerSecret_Description (18.59s)
=== RUN   TestAccAwsSecretsManagerSecret_KmsKeyID
--- PASS: TestAccAwsSecretsManagerSecret_KmsKeyID (52.88s)
=== RUN   TestAccAwsSecretsManagerSecret_RotationLambdaARN
--- PASS: TestAccAwsSecretsManagerSecret_RotationLambdaARN (47.56s)
=== RUN   TestAccAwsSecretsManagerSecret_RotationRules
--- PASS: TestAccAwsSecretsManagerSecret_RotationRules (46.72s)
=== RUN   TestAccAwsSecretsManagerSecret_Tags
--- PASS: TestAccAwsSecretsManagerSecret_Tags (34.72s)
=== RUN   TestAccAwsSecretsManagerSecretVersion_Basic
--- PASS: TestAccAwsSecretsManagerSecretVersion_Basic (11.76s)
=== RUN   TestAccAwsSecretsManagerSecretVersion_VersionStages
--- PASS: TestAccAwsSecretsManagerSecretVersion_VersionStages (31.19s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	326.005s
```

### CODE REVIEW

File: aws/resource_aws_secretsmanager_secret.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+
```
Comment (paultyng) [inline aws/resource_aws_secretsmanager_secret.go]: whitespace

File: aws/resource_aws_secretsmanager_secret.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"automatically_after_days": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+			"tags": {
```
Comment (paultyng) [inline aws/resource_aws_secretsmanager_secret.go]: https://github.com/terraform-providers/terraform-provider-aws/blob/44af183718e5ddf44de518a04fd858f84cb350df/aws/tags.go#L20:6

should this use the common schema?

File: aws/resource_aws_secretsmanager_secret.go (hunk 1 above)
Comment (bflad) [reply aws/resource_aws_secretsmanager_secret.go]: `goimports` strikes again! Not sure why it thinks it needs to do this sometimes. I'll get it updated

File: aws/resource_aws_secretsmanager_secret.go (hunk 2 above)
Comment (bflad) [reply aws/resource_aws_secretsmanager_secret.go]: Sure! Will update. 👍

**Comment by bflad** [comment]
This has been released in [version 1.16.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1160-april-25-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**